### DISCUSSION TITLE
How to delete unwanted feature branch's collection from DB2

### DISCUSSION THREAD

**Discussion by rahulrp1** [IBM/dbb-zappbuild#464] [Q&A]
Hi Team,

How to delete unwanted feature branch's collection from DB2 Database ?
**Usecase :**
 We are following workflow in which we are creating feature branch for new features /improvement .Once developer worked upon it and deploy it to respective test environments ,we are merging this feature branch to develop branch and deleting feature branch and proceeding to next release activites. In this case ,as feature branch is deleted ,we no more want the collection of that feature branch.
How can to delete this unwanted feature collection from db2 database as per dbb 2.0.0?

Let me know incase need any more details.

**Comment by dennis-behm** [comment]
@rahulrp1 , 

are you aware of the DBB CLI to interact with the Metadatastore - https://www.ibm.com/docs/en/dbb/2.0?topic=interface-collection-commands ? This CLI allows you to delete the collection, with the CLI to the build result groups, also the build result group.

Alternatively, you can write a script similar to https://github.com/IBM/dbb/tree/dbb_1_x/Utilities/WebAppCleanUp that you can be included into a pipeline job. For instance you can implement a process to ru every night to check the existing collections and compare it with the available feature branches in your git provider.

**Reply by rahulrp1** [reply]
Thanks @dennis-behm , I will check the provided source and I will get back to you in case need any further assistance.

**Reply by nixpix49** [reply]
@rahulrp1, 

We are using the WebAppCleanUp-utility with dbb 1.1.4 today. Since we are running an old version of dbb it's ok.
But as soon as we start using dbb 2.0 with db2 that utility won't work anymore. Will look into it in the near future (I hope).
So you are not alone, we have sort of the same idea.

Have a nice day!

Regards
Nikos

**Reply by nixpix49** [reply]
Just a note, build.groovy can be executed with the --reset parameter set.
// handle -r,--reset option
if (props.reset && props.reset.toBoolean())  {
  println("** Reset option selected")
  println("* Deleting collection ${props.applicationCollectionName}")
  metadataStore.deleteCollection(props.applicationCollectionName)
  println("* Deleting collection ${props.applicationOutputsCollectionName}")
  metadataStore.deleteCollection(props.applicationOutputsCollectionName)
  println("* Deleting build result group ${props.applicationBuildGroup}")
  metadataStore.deleteBuildResults(props.applicationBuildGroup)
  System.exit(0)
}
In the test-folder dbb-zappbuild/test/testScripts you also have resetBuild.groovy which executes the above mentioned build.groovy script with the --reset parameter set, so collections/buildresult deleted... 
Will have a look at this one and adapt to our needs...

**Reply by rahulrp1** [reply]
@nixpix49  Thanks , I will check this.
